source/puppet/4.8/env_isolation.md

```diff
@@ -8,25 +8,25 @@ description: "Generating metadata to isolate resources in environments in Puppet
 [code_mgr_env]: {{pe}}/code_mgr.html#environment-isolation-metadata
 
 
-If you have multiple environments, environment isolation prevents resource types from leaking between your various environments. To use environment isolation, you'll generate metadata files that Puppet can use in place of the normal Ruby resource type implementations.
+If you have multiple environments, environment isolation prevents module resource types from leaking between your various environments. To use environment isolation, you'll generate metadata files that Puppet can use in place of the normal Ruby resource type implementations.
 
 In open source Puppet, enable environment isolation with the `generate types` command. In Puppet Enterprise, environment isolation is automatically provided by [Code Manager][code_mgr_env]. Environment isolation is not supported for r10k with Puppet Enterprise.
 
 ## Preventing resource types leaks in multiple environments
 
-If you use multiple environments with Puppet, you might encounter issues with multiple versions of the same resource type leaking between your various environments. 
+If you use multiple environments with Puppet, you might encounter issues with multiple versions of the same resource type leaking between your various environments. This happens with custom resource types in modules, not with Puppet's built-in resource types.
```

Comment:
"custom resource types in modules" as opposed to what other resource types? Non custom not in modules, custom not in modules  what if they are in a gem? It can happen for all but the resource types in puppet itself.